File: 179-largest-number/largest-number.java
class Solution {
    public String largestNumber(int[] nums) {
        String[] asStr = new String[nums.length];

        for(int i = 0; i < nums.length; i++){
            asStr[i] = String.valueOf(nums[i]);
        }

        Arrays.sort(asStr, (a, b) -> (b + a).compareTo(a + b));

        if(asStr[0].equals("0")){
            return "0";
        }

        StringBuilder largestNum = new StringBuilder();

        for(String numAsStr : asStr){
            largestNum.append(numAsStr);
        }

        return largestNum.toString();
    }
}

File: 179-largest-number/largest-number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        nums = [str(num) for num in nums]

        nums.sort(key=cmp_to_key(lambda a, b: -1 if a+b>b+a else 1))

        if nums[0]=='0':
            return "0"

        else:
            return "".join(nums)